src/app/stack/card-details/component-information/component-information.component.html

```diff
@@ -1,36 +1,81 @@
             <div class="main-content">
                 <div [class]="position?.className" *ngFor="let position of positions;">
                     <div [ngSwitch]="position?.identifier">
                         <!-- Security Card -->
                         <div *ngSwitchCase="'serial'">
                             <!-- Serial Numbers -->
-                            <span [innerText]="serial + 1"></span>
+                            <span [innerText]="serial + 1 < 10 ? '#0' + (serial + 1) : '#' + (serial + 1)"></span>
                         </div>
                         <div *ngSwitchCase="'component'">
                             <!-- Component Names -->
-                                <span [innerText]="comp?.name"></span>
-                                <span class="recommendation-reason" [tooltip]="component?.reason" *ngIf="component?.reason && type === 'recommendation' && comp?.name">Why this dependency?</span>
+                            <a class="name" [href]="generateUrl(comp?.url)" target="_blank" title="command/ctrl + click" [innerText]="comp?.name"></a>
+                            <span class="recommendation-reason" [tooltip]="component?.reason"
+                                *ngIf="component?.reason && type === 'recommendation' && comp?.name">Why this
+                                dependency?</span>
                             <!--
                                 <reco-name [name]="comp?.name" [reason]="comp?.recommendation?.reason" *ngIf="comp?.recommendation && (comp?.recommendation?.type === 'companion')"></reco-name>
                             -->
                         </div>
+
                         <div *ngSwitchCase="'cveCount'">
                             <!-- cveCount -->
-                            <span [innerText]="comp?.securityDetails && comp?.securityDetails.totalIssues || 0"></span>
+                            <span
+                                *ngIf="tabType == 'private' && comp?.private_vulnerabilities && comp?.private_vulnerabilities.length > 0"
+                                [innerText]="comp?.privateSecurityDetails && comp?.privateSecurityDetails.totalIssues" class="count"></span>
+                            <span
+                                *ngIf="tabType == 'public' && comp?.public_vulnerabilities && comp?.public_vulnerabilities.length > 0"
+                                [innerText]="comp?.publicSecurityDetails && comp?.publicSecurityDetails.totalIssues" class="count"></span>
+                            <span
+                                *ngIf="tabType == 'public' && comp?.public_vulnerabilities && comp?.public_vulnerabilities.length == 0"
+                                [innerText]="'Check Transitives'"></span>
+                            <span
+                                *ngIf="tabType == 'private' && comp?.private_vulnerabilities && comp?.private_vulnerabilities.length == 0"
+                                [innerText]="'Check Transitives'"></span>
```

Comment:
Since these are related to transitives, should these be nested inside the parent span?

Reply:
Yes, I had tried it out with the if-else statement but i did not work out. 
So I kept it this way.